read-fonts/src/tables/varc.rs

```diff
@@ -8,21 +8,230 @@ pub use super::{
     pub fn axis_indices(&self, nth: usize) -> Result<PackedDeltas, ReadError> {
-        let Some(axis_indices_list) = self.axis_indices_list() else {
-            return Err(ReadError::InvalidCollectionIndex(nth as u32));
+        let raw = self.axis_indices_list().nth(nth)?;
+        Ok(PackedDeltas::consume_all(raw.into()))
+    }
+
+    /// Friendlier accessor than directly using raw data via [Index2]
+    ///
+    /// nth would typically be obtained by looking up a [GlyphId] in [Self::coverage].
+    pub fn glyph(&self, nth: usize) -> Result<VarcGlyph<'_>, ReadError> {
+        let raw = Some(self.var_composite_glyphs()).nth(nth)?;
+        Ok(VarcGlyph {
+            table: self,
+            data: raw.into(),
+        })
+    }
+}
+
+/// I'm just a happy little blob full of components
+pub struct VarcGlyph<'a> {
+    table: &'a Varc<'a>,
+    data: FontData<'a>,
+}
+
+impl<'a> VarcGlyph<'a> {
+    /// <https://github.com/fonttools/fonttools/blob/5e6b12d12fa08abafbeb7570f47707fbedf69a45/Lib/fontTools/ttLib/tables/otTables.py#L404-L409>
+    pub fn components(&self) -> impl Iterator<Item = Result<VarcComponent<'a>, ReadError>> {
+        VarcComponentIter {
+            table: self.table,
+            cursor: self.data.cursor(),
+        }
+    }
+}
+
+struct VarcComponentIter<'a> {
+    table: &'a Varc<'a>,
+    cursor: Cursor<'a>,
+}
+
+impl<'a> Iterator for VarcComponentIter<'a> {
+    type Item = Result<VarcComponent<'a>, ReadError>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.cursor.is_empty() {
+            return None;
+        }
+        Some(VarcComponent::parse(self.table, &mut self.cursor))
+    }
+}
+
+#[allow(dead_code)] // TEMPORARY
+pub struct VarcComponent<'a> {
+    flags: VarcFlags,
+    gid: GlyphId,
+    condition_index: Option<u32>,
+    axis_indices_index: Option<u32>,
+    axis_values: Option<PackedDeltas<'a>>,
+    axis_values_var_index: Option<u32>,
+    transform_var_index: Option<u32>,
+    dx: i16,
+    dy: i16,
+    rotation: f32,
+    sx: f32,
+    sy: f32,
+    skewx: f32,
+    skewy: f32,
+    center_x: i16,
+    center_y: i16,
+}
+
+impl<'a> VarcComponent<'a> {
+    /// Requires access to VARC fields to fully parse.
+    ///
+    ///  * HarfBuzz [VarComponent::get_path_at](https://github.com/harfbuzz/harfbuzz/blob/0c2f5ecd51d11e32836ee136a1bc765d650a4ec0/src/OT/Var/VARC/VARC.cc#L132)
+    // TODO: do we want to be able to parse into an existing glyph to avoid allocation?
+    fn parse(table: &Varc, cursor: &mut Cursor<'a>) -> Result<Self, ReadError> {
+        let raw_flags = cursor.read_u32_var()?;
+        let flags = VarcFlags::from_bits_truncate(raw_flags);
+        // Ref https://github.com/harfbuzz/boring-expansion-spec/blob/main/VARC.md#variable-component-record
+
+        // This is a GlyphID16 if GID_IS_24BIT bit of flags is clear, else GlyphID24.
+        let gid = if flags.contains(VarcFlags::GID_IS_24BIT) {
+            let gid = cursor.read_be::<Uint24>()?.get().to_u32();
```

Comment:
you can use `read` instead of `read_be`, the only difference is that `read_be` returns `BigEndian<T>` and `read` returns `T` (handling the conversion). `read_be` only exists for codegening certain records that we can't zerocopy because they have variable size, but the types expect the `BigEndian` wrapper.

can open an issue to clarify this on the docs of `read_be`.

Reply:
Can confirm I was initially unclear whether I could safely use read or not. I think I concluded I could and updated to do so. Updated _almost_ everything I suppose. Will update.